app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,438 @@
+                connect(id, ConstraintSet.START, sectionTitleTextView.id, ConstraintSet.START)
+                connect(id, ConstraintSet.TOP, sectionTitleTextView.id, ConstraintSet.BOTTOM, DEFAULT_COLLAPSIBLE_SUBTITLE_TOP_MARGIN)
+            }
+            with(lineSeparatorView) {
+                connect(
+                    id,
+                    ConstraintSet.START,
+                    sectionSubtitleTextView.id,
+                    ConstraintSet.START,
+                    COLLAPSIBLE_LINE_SEPARATOR_START_MARGIN
+                )
+                connect(id, ConstraintSet.BOTTOM, containerFrameLayout.id, ConstraintSet.TOP)
+            }
+            with(containerFrameLayout) {
+                connect(id, ConstraintSet.END, ConstraintSet.PARENT_ID, ConstraintSet.END)
+                connect(id, ConstraintSet.START, ConstraintSet.PARENT_ID, ConstraintSet.START)
+                connect(id, ConstraintSet.BOTTOM, footerConstraintLayout.id, ConstraintSet.TOP)
+            }
+            with(footerConstraintLayout) {
+                connect(id, ConstraintSet.END, ConstraintSet.PARENT_ID, ConstraintSet.END)
+                connect(id, ConstraintSet.START, sectionTitleTextView.id, ConstraintSet.START)
+                connect(id, ConstraintSet.BOTTOM, ConstraintSet.PARENT_ID, ConstraintSet.BOTTOM)
+            }
+            applyTo(sectionConstraintLayout)
+        }
+
+        ConstraintSet().apply {
+            clone(this@CollapsibleView)
+
+            with(sectionCardView) {
+                connect(
+                    id,
+                    ConstraintSet.TOP,
+                    ConstraintSet.PARENT_ID,
+                    ConstraintSet.BOTTOM
+                )
+                connect(id, ConstraintSet.BOTTOM, this@CollapsibleView.id, ConstraintSet.BOTTOM)
+                connect(
+                    id,
+                    ConstraintSet.START,
+                    ConstraintSet.PARENT_ID,
+                    ConstraintSet.START,
+                    COLLAPSIBLE_CARD_VIEW_START_MARGIN
+                )
+                connect(
+                    id,
+                    ConstraintSet.END,
+                    this@CollapsibleView.id,
+                    ConstraintSet.END,
+                    COLLAPSIBLE_CARD_VIEW_END_MARGIN
+                )
+            }
+            applyTo(this@CollapsibleView)
+        }
+
+        attrs?.let { setupAttrs(it) }
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        if (collapsibleViewId != NO_ID) {
+            (context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as? LayoutInflater)
+                ?.inflate(collapsibleViewId, null, false)?.let {
+                    setContent(it)
+                }
+
+            (context as? AppCompatActivity)?.windowManager?.defaultDisplay?.let { display ->
+                containerFrameLayout.measure(display.width, display.height)
```

Comment:
display.width and display.height are deprecated

Reply:
Solved